### PR TITLE
Attendance SLAP

### PR DESCRIPTION
**Pull request by benchan911**
improve code quality

### CODE REVIEW

File: src/main/java/seedu/command/attendance/AddAttendanceList.java [hunk 1]
```diff
@@ -39,77 +45,101 @@ public void addToList() throws PacException {
...
+            studentName = ui.getUserInput();
+            if (studentName.equals("done")) {
+                break;
+            }
+            if (attendanceList.isDuplicate(studentName)) {
+                throw new PacException("Duplicated name found, student name : [ " + studentName + " ] not added");
+            } else {
+                UI.display("To mark the student as present, please use 'y' or 'Y'.");
+                UI.display("By default the student will be marked as absent,"
                         + " if any other input is given.");
```
Comment (Anqi-nus) [inline src/main/java/seedu/command/attendance/AddAttendanceList.java:122]: Might want to add a method in UI to display this two sentences. then use ui...() to look neater.

File: src/main/java/seedu/command/attendance/AddAttendanceList.java [hunk 2]
```diff
@@ -39,77 +45,101 @@ public void addToList() throws PacException {
...
+
+    /**
+     * Method to create new attendanceList using a line-by-line input from the user.
+     * @throws PacException If parameter provided is invalid.
+     */
+    private void lineByLine() throws PacException {
+        while (!studentName.equals("done")) {
+            UI.display("Please key in student name.");
+            ui.readUserInput();
+            studentName = ui.getUserInput();
```
Comment (Anqi-nus) [inline src/main/java/seedu/command/attendance/AddAttendanceList.java:113]: might want to combine these three lines to one method getStudentName()

File: src/main/java/seedu/command/attendance/AddAttendanceList.java [hunk 3]
```diff
@@ -39,77 +45,101 @@ public void addToList() throws PacException {
...
-                ui.readUserInput();
-                input = ui.getUserInput();
-                if (input.equals("done")) {
-                    break;
-                }
-                attendances.addToList(attendanceParser.parseAttendance(input), eventName);
-                newStudentList.addToList(name);
-                studentNumber++;
+            ui.readUserInput();
+            input = ui.getUserInput();
```
Comment (Anqi-nus) [inline src/main/java/seedu/command/attendance/AddAttendanceList.java:91]: These three lines as well.

Comment (Anqi-nus) [reply src/main/java/seedu/command/attendance/AddAttendanceList.java:91]: Refer to comment below

File: src/main/java/seedu/command/interpreter/AttendanceCommandInterpreter.java [hunk 4]
```diff
@@ -65,43 +68,50 @@ public Command decideCommand(String commandDescription) throws PacException {
         logger.info("My First Log");
         logger.fine("My Second Log");
 
-        switch (commandType) {
+        switch (commandType.toLowerCase()) {
```
Comment (Anqi-nus) [inline src/main/java/seedu/command/interpreter/AttendanceCommandInterpreter.java]: I use toLowerCase().trim() for all input now, might want to consider that also.

File: src/main/java/seedu/command/interpreter/AttendanceCommandInterpreter.java [hunk 5]
```diff
@@ -115,4 +125,24 @@ public Command decideCommand(String commandDescription) throws PacException {
     private AttendanceList getAttendance(String eventName) throws PacException {
         return eventList.getEvent(eventName).getAttendanceList();
     }
+
+    private boolean sortByName() {
+        UI.display("Please Key in either 'name' or 'status'.");
+        ui.readUserInput();
+        String sortType = ui.getUserInput();
```
Comment (Anqi-nus) [inline src/main/java/seedu/command/interpreter/AttendanceCommandInterpreter.java]: combine

File: src/main/java/seedu/command/interpreter/AttendanceCommandInterpreter.java [hunk 6]
```diff
@@ -115,4 +125,24 @@ public Command decideCommand(String commandDescription) throws PacException {
...
+        if (sortType.toLowerCase().equals("name")) {
+            return true;
+        }
+        return false;
+    }
+
+    private boolean sortByStatus() {
+        UI.display("Please Key in either 'name' or 'status'.");
+        ui.readUserInput();
+        String sortType = ui.getUserInput();
```
Comment (Anqi-nus) [inline src/main/java/seedu/command/interpreter/AttendanceCommandInterpreter.java]: combine

**Review by Anqi-nus — COMMENTED** [review]
Generally fine. I think some lines can still be extracted out to look neater.

File: src/main/java/seedu/attendance/AttendanceList.java [hunk 7]
```diff
@@ -16,33 +16,67 @@
 public class AttendanceList {
 
     protected ArrayList<Attendance> attendanceList;
```
Comment (lowjiayee) [inline src/main/java/seedu/attendance/AttendanceList.java:18]: This attribute doesn't seem to be used at all... Perhaps can consider removing it?

Comment (benchan911) [reply src/main/java/seedu/attendance/AttendanceList.java:18]: it is used, in fetching the attendance list.

File: src/main/java/seedu/command/interpreter/AttendanceCommandInterpreter.java (hunk 4 above)
Comment (benchan911) [reply src/main/java/seedu/command/interpreter/AttendanceCommandInterpreter.java]: Will include that, thankyou!

File: src/main/java/seedu/command/attendance/AddAttendanceList.java (hunk 1 above)
Comment (benchan911) [reply src/main/java/seedu/command/attendance/AddAttendanceList.java:122]: Made the changes to combine

**Pull request merged**